src/IStakeManager.sol

```diff
@@ -0,0 +1,35 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.18;
+
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { ITrustedCodehashAccess } from "./access/ITrustedCodehashAccess.sol";
+
+interface IStakeManager is ITrustedCodehashAccess {
+    error StakeManager__FundsLocked();
+    error StakeManager__InvalidLockTime();
+    error StakeManager__InsufficientFunds();
+    error StakeManager__StakeIsTooLow();
+
+    function stake(uint256 _amount, uint256 _seconds) external;
+    function unstake(uint256 _amount) external;
+    function lock(uint256 _secondsIncrease) external;
+    function leave() external returns (bool _leaveAccepted);
+    function acceptUpdate() external returns (address _migrated);
+
+    function potentialMP() external view returns (uint256);
+    function totalMP() external view returns (uint256);
+    function totalStaked() external view returns (uint256);
+    function totalSupply() external view returns (uint256 _totalSupply);
+    function totalSupplyMinted() external view returns (uint256 _totalSupply);
+    function pendingReward() external view returns (uint256);
+    function getStakedBalance(address _vault) external view returns (uint256 _balance);
```

Comment:
I think here we can drop some as well.

`pendingReward()` is not necessary as the won't be any rewards.
`totalSupplyMinted()` currently says `totalStaked + totalMP + potentialMP`. I don't think we want that. `potentialMP` is the maximum MP that can exist in the system based on the stake amount (it's basicallly `maxMP` from the other system).

So we only need:

- `potentialMP` (and `userPotentialMP` I guess)
- `totalMP` (all minted MPs)
- `totalStaked` (all staked balance)
- `totalSupply` (totalStaked + totalMP)

Reply:
> pendingReward() is not necessary as the won't be any rewards.

I would leave it here, because the contract being updated STILL have rewarding. Once the rewarding is fully removed, we can remove it there afterwards. 

> potentialMP (and userPotentialMP I guess) 

This shouldn't be part of the interface because is not common to StakeManager.sol


> totalSupply (totalStaked + totalMP)

I'll update the logic to remove potentialMPs from here. And seems like  totalSupplyMinted() should be dropped from the interface because its something related just to StakeManager? Or maybe we can change it so:

`totalSupply() => totalStaked + totalMP + calcMP(totalStaked, block.timestamp - lastMPUpdatedTime)`

Reply:
At the end it depends on what we want `totalSupply()` to represent.
Is it the actual supply that is in existence (`totalMP + totalStaked`) or is it whatever is in existence + whatever there is left to mint as per `potentialMP`

@gravityblast what are your thougths on this

Reply:
`totalSupply() => totalStaked + totalMP + calcMP(totalStaked, block.timestamp - lastMPUpdatedTime)` is actually wrong, but its something like that as its not considering potentialMP, but it should be something like that.

The scenario is:

For an external system, which wants to read the "current balance" of a user, and compare for how much of "total supply" exists, it would want that both values, for user balance, and for total supply, to represent how much it would have if the contract is updated to last mint. 

If we dont have this, than, an external UI would not be able to read the values as if the contact would be updated. It would always have the values from last mint.

Reply:
Yea very good point. The `totalSupply` in original StakeManager I think is a relict from the past where we weren't yet sure if the token will be represented by the contract as well.

Now that we're working on xp-token contract, we'll have a dedicated `totalSupply` there.